### ISSUE TITLE
Some ARAs Currently Returning 0 Results for Chemicals to Gene

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/testing#10]
**Query:** https://github.com/NCATSTranslator/testing/blob/main/ars-requests/not-none/spmsyChemicals.json 
**Parent PK:**  d187edcd-f4a1-42e8-9248-eee0cb161fdb

![image](https://user-images.githubusercontent.com/38321826/110952771-978b3780-8314-11eb-9a2d-aff9e5740775.png)

**Results not being returned from:**  BTE (Exploring Agent), Genetics KP, MolePro, Explanatory Agent, Improve Agent, Text Mining KP
Other KPs?

**Results are being returned from:**  ARAGORN (Ranking Agent), ARAX (Expander Agent), Unsecret Agent, CAM KP (Exposures Provider)

**Comment by kevinxin90** [comment]
The Exploring Agent has fixed the issue. Feel free to try again.

**Comment by sstemann** [comment]
@vdancik looks like the HGNC identifier returns 38 results from MolePro now:  dcd99eff-e0d6-4b4a-9a8f-ffa744bff715
@brettasmi looks like the query update to HGNC id 11123 didn't work for Improving Agent 340bdf01-6cc3-4d9c-8d3f-1150ddd74c5c

**Comment by sstemann** [comment]
> The Exploring Agent has fixed the issue. Feel free to try again.

Thanks @kevinxin90 !  looks like ARS is getting 128 results from BTE now 🎉

**Comment by brettasmi** [comment]
Yes, that's expected until we make a fix to look for the NCBIGENE over NCBIGene internally. I will update when resolved. The new query is as specified below, correct?

```
{
  "message": {
    "query_graph": {
      "nodes": {
        "n0": { "id": "HGNC:11123", "category": "biolink:Gene" },
        "n1": { "category": "biolink:ChemicalSubstance" }
      },
      "edges": { "e01": { "subject": "n0", "object": "n1" } }
    }
  },
  "callback": "http://localhost:8000/ars/api/messages/340bdf01-6cc3-4d9c-8d3f-1150ddd74c5c"
}
```

**Comment by MarkDWilliams** [comment]
Brett, the query you posted is a variant that I tried in order to test with the HGNC identifier, but I believe the one in the testing repo will remain in NCBIGENE form

**Comment by brettasmi** [comment]
After updating the internal configuration, this should work with imProving Agent.

**Comment by sstemann** [comment]
> After updating the internal configuration, this should work with imProving Agent.

@brettasmi looks like it's working per today's standup - leaving a note here for you to run the reasonervalidator on the response.

**Comment by sstemann** [comment]
@pg427 when Explanatory Agent has an update, please post here.

**Comment by pg427** [comment]
@sstemann Understood. The Explanatory Agent is at the moment discussing a workflow to implement the relation. We will have further updates on this moving forward, detailing this association's implementation progress.

**Comment by sstemann** [comment]
For original query, UniProtKB:P52788, PK: f70b81d9-9b40-4347-82cf-baaa13f93048 as of 4/20/21

![image](https://user-images.githubusercontent.com/38321826/115443170-eca64d00-a1e0-11eb-8544-4bfd1101ff3e.png)

**Comment by colleenXu** [comment]
Note that using an updated query like this, BTE can find answers linked to either the Protein or the Gene. 

![Screen Shot 2021-09-21 at 11 26 03 PM](https://user-images.githubusercontent.com/43731687/134293439-48843fe6-4f17-4326-a8a6-3b4c167e6051.png)

```
{
    "message": {
        "query_graph": {
            "nodes": {
                "n0": {
                    "ids": ["UniProtKB:P52788", "HGNC:11123"],
		            "categories":["biolink:Gene", "biolink:Protein"]
                },
                "n1": {
                    "categories": ["biolink:ChemicalEntity"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "n0",
                    "object": "n1"
                }
            }
        }
    }
}
```

**Comment by sstemann** [comment]
the PK for this query is:  fe6bb9c7-df18-4541-adf1-98d405868e5f

i updated the linked query to ChemicalEntity

![image](https://user-images.githubusercontent.com/38321826/136048972-5c9b437c-9754-43f6-9389-9416dd5e859d.png)

looks like ARAGORN and Unsecret responded to the original query

**Comment by cbizon** [comment]
@sstemann  Can you point to the new query?   I'm probably missing it, but I don't see it linked above...

**Comment by sstemann** [comment]
@cbizon sure i uploaded it https://github.com/NCATSTranslator/testing/blob/main/ars-requests/not-none/1.2/GH-10-proteinSpmsyChemicals.json

**Comment by cbizon** [comment]
Thanks.  I suspect you caught us last night when there was maintenance going on.  Reran it just now and it seems like we returned results https://arax.ncats.io/?r=a1f1cfde-605b-42a5-9c5e-234c1b91f7e5
